connect.go

```diff
@@ -243,6 +253,21 @@ type Spec struct {
 	IsClient   bool   // otherwise we're in a handler
 }
 
+// Peer describes the other party to an RPC. When accessed client-side, Addr
+// contains the host or host:port from the server's URL. When accessed
+// server-side, Addr contains the client's address in IP:port format.
+type Peer struct {
+	Addr string
```

Comment:
gRPC exposes this as `net.Addr` (which includes a Network string: https://pkg.go.dev/google.golang.org/grpc/peer#Peer). Do you think we won't need to provide that additional data?

Reply:
Possibly? We can always expose a `Network` attribute down the road too. I'm not sure how to reliably detect non-TCP transports through the net/http abstractions - for example, unix domain sockets are sort of the wild west, and I have no idea what HTTP/3 will end up looking like.